wallet-new/src/Cardano/Wallet/WalletLayer/Kernel/Wallets.hs

```diff
@@ -148,3 +152,21 @@ getWallets :: Kernel.DB -> IxSet V1.Wallet
 getWallets db = IxSet.fromList . map (toWallet db) . IxSet.toList $ allRoots
   where
     allRoots = db ^. dbHdWallets . HD.hdWalletsRoots
+
+-- | Gets Utxos per account of a wallet.
+getWalletUtxos
+    :: V1.WalletId
+    -> Kernel.DB
+    -> Either GetUtxosError [(V1.Account, Utxo)]
+getWalletUtxos wid db = runExcept $ do
+    accounts <- withExceptT GetUtxosErrorFromGetAccountsError $
+        exceptT $ getAccounts wid db
```

Comment:
Sorry missed this. This should not call `getAccounts`. That's using the V1 API to get access to the wallet, which is a lossy query. There's no need for that. Instead use `fromRootId` to translate the `WalletId` to a `HdRootId` (that may fail with a decoding error, which is why the first error condition is still needed), and then use `accountsByRootId` from `Cardano.Wallet.Kernel.DB.Read` to get the `IxSet` of accounts.